### PR TITLE
Feature/bbb 866 mobility aids

### PR DESCRIPTION
**Pull request by pauljroberts**

### CODE REVIEW

File: src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    mockMvc
+        .perform(
+            post("/add-mobility-aid")
+                .param("id", "1234")
+                .contentType("application/x-www-form-urlencoded")
+                .sessionAttr("JOURNEY", journey))
+        .andExpect(status().isFound())
+        .andExpect(redirectedUrl("/add-mobility-aid"))
+        .andExpect(flash().attribute("formRequest", form));
```
Comment (mgilgar) [inline src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java]: I think this code is not checking if we are generating the correct error messages. Please check 'WhereCanYouWalkControllerTest.java'

  @Test
  public void submit_givenEmptyValues_thenShouldDisplayNotBlankValidationErrors() throws Exception {
    when(mockRouteMaster.redirectToOnSuccess(any(WhereCanYouWalkForm.class), any(Journey.class)))
        .thenReturn("redirect:/testSuccess");

    mockMvc
        .perform(
            post("/where-can-you-walk")
                .param("destinationToHome", "")
                .param("timeToDestination", "")
                .sessionAttr("JOURNEY", journey))
        .andExpect(status().isFound())
        .andExpect(redirectedUrl("/someValidationError"));

    ArgumentCaptor<BindingResult> captor = ArgumentCaptor.forClass(BindingResult.class);
    verify(mockRouteMaster, times(1))
        .redirectToOnBindingError(any(), any(), captor.capture(), any());
    BindingResult bindingResult = (BindingResult) captor.getValue();

    assertThat(bindingResult.getErrorCount()).isEqualTo(2);
    assertTrue(containsError(bindingResult.getFieldErrors(), "destinationToHome", "", "NotBlank"));
    assertTrue(containsError(bindingResult.getFieldErrors(), "timeToDestination", "", "NotBlank"));
  }


There may be more beautiful ways to do it but it checks the errors.

File: src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    mockMvc
+        .perform(
+            post("/add-mobility-aid")
+                .param("aidType", "WALKING_AID")
+                .param("howProvidedCodeField", "PRESCRIBE")
+                .param("usage", "usage")
+                .contentType("application/x-www-form-urlencoded")
+                .sessionAttr("JOURNEY", journey))
+        .andExpect(status().isOk())
```
Comment (mgilgar) [inline src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java]: I think as Stephen said, this should be an isFound and so on. Please refer to my previous comment about how to check the specific validation errors.

File: src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java (hunk 1 above)
Comment (pauljroberts) [reply src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java]: Is checking error message, but worth discussing better approach.

File: src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddController.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    if (!model.containsAttribute(FORM_REQUEST)) {
+      model.addAttribute(FORM_REQUEST, new MobilityAidAddForm());
+    }
+
+    model.addAttribute(
+        "howProvidedOptions",
+        new RadioOptionsGroup.Builder()
+            .titleIsLabel()
+            .titleMessageKeyApplicantAware("mobilityaid.howprovided.optiongroup.title", journey)
+            .addOption(HowProvidedCodeField.PRESCRIBE, "mobilityaid.howprovided.option.prescribed")
```
[the remark this answers is not in the document]
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddController.java]: That's not the convention in any pages I've done :)

File: src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddController.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  @PostMapping
+  public String submit(
+      @ModelAttribute(JOURNEY_SESSION_KEY) Journey journey,
+      @Valid @ModelAttribute(FORM_REQUEST) MobilityAidAddForm mobilityAidAddForm,
+      BindingResult bindingResult,
+      RedirectAttributes attr) {
+
+    if (MobilityAidAddForm.AidType.WALKING_AID == mobilityAidAddForm.getAidType()
```
[the remark this answers is not in the document]
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddController.java]: This is the pattern I have taken from some other controller - can't remember which one.  If you want to show me an example of doing it in bean then that is fine.

File: src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java (hunk 1 above)
Comment (pauljroberts) [reply src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/MobilityAidAddControllerTest.java]: Better approach applied.

**Pull request merged**